cmp/empty.go

```diff
@@ -0,0 +1,90 @@
+package cmp
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/go-spatial/geom"
+)
+
+func IsEmptyPoint(pt [2]float64) bool {
+	return pt != pt
+}
+
+func IsEmptyPoints(pts [][2]float64) bool {
+	for _, v := range pts {
+		if !IsEmptyPoint(v) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func IsEmptyLines(lns [][][2]float64) bool {
+	for _, v := range lns {
+		if !IsEmptyPoints(v) {
+			return false
+		}
+	}
+
+	return true
+}
+
+func IsNil(a interface{}) bool {
+	defer func() { recover() }()
+	return a == nil || reflect.ValueOf(a).IsNil()
+}
+
+func IsEmptyGeo(geo geom.Geometry) (isEmpty bool, err error) {
+	if IsNil(geo) {
+		return true, nil
+	}
+
+	switch g := geo.(type) {
+	case [2]float64:
+		return IsEmptyPoint(g), nil
+
+	case geom.Pointer:
+		return IsEmptyPoint(g.XY()), nil
+
+	case [][2]float64:
+		return IsEmptyPoints(g), nil
+
```

Comment:
Maybe add `[]geom.Point` and `[]geom.Pointer` to this list as we have `[][2]float64`